mtags/src/main/scala/scala/meta/internal/pc/ScaladocCompletion.scala

```diff
@@ -0,0 +1,199 @@
+      // """
+      val builder = new StringBuilder()
+
+      val hasConstructor = associatedDef.symbol.primaryConstructor.isDefined && !associatedDef.symbol.isAbstractClass
+      val hasReturnValue =
+        associatedDef.symbol.isMethod &&
+          !(associatedDef.symbol.asMethod.returnType.finalResultType
+            =:= typeOf[Unit])
+
+      // newline after `/**`
+      builder.append("\n")
+
+      // * $0 <- move cursor here.
+      builder.append(s"${indent}*")
+      if (clientSupportsSnippets) builder.append(" $0\n")
+      else builder.append("\n")
+
+      // add empty line if there's parameter or "@return" or "@constructor" line.
+      if (params.nonEmpty || hasReturnValue || hasConstructor)
+        builder.append(s"${indent}*\n")
+
+      // * @constructor
+      if (hasConstructor) builder.append(s"${indent}* @constructor\n")
+
+      // * @param p1
+      // * @param p2
+      params.foreach(param => builder.append(s"${indent}* @param ${param}\n"))
+
+      // * @return
+      // */
+      if (hasReturnValue) builder.append(s"${indent}* @return\n")
+      builder.append(s"${indent}*/")
+
+      List(
+        new TextEditMember(
+          "Scaladoc Comment",
+          new l.TextEdit(
+            editRange,
+            builder.toString()
+          ),
+          completionsSymbol(associatedDef.name.toString()),
+          label = Some("/** */"),
+          detail = Some("Scaladoc Comment")
+        )
+      )
+    }
+
+    // Infers the indentation at the completion position by counting the number of leading
+    // spaces in the line.
+    // For example:
+    // |"""
+    // |object A {
+    // |  /**<COMPLETE> // inferred indent is 4 spaces
+    // |  def foo(x: Int) = ???
+    // |}
+    // |"""
+    private def inferIndent(pos: Position): String = {
+      if (metalsConfig.snippetAutoIndent()) {
+        ""
+      } else {
+        // 3 is the length of "/**$" <- $ is a cursor
+        " " * (pos.column - 4)
+      }
+    }
+
+    /**
+     * Returns the parameter names of the given memberDef.
+     *
+     * @param memberDef The memberDef to construct scaladoc.
+     */
+    private def getParams(memberDef: MemberDef): List[String] = {
+      memberDef match {
+        case defdef: DefDef =>
+          defdef.symbol.paramss.flatten
+            .filter(!_.isSynthetic)
+            .map(_.name.toString())
+        case clazz: ClassDef =>
+          // If the associated def is a class definition,
+          // retrieve the constructor from the class, and caluculate the lines
+          // from the constructor definition instead.
+          clazz.symbol.primaryConstructor.paramss.flatten
+            .filter(sym =>
+              // FIXME: All synthetic params should be filtered out by checking SYNTHETIC flag.
+              // evidence params for classs constructor look like they don't have SYNTHETIC flag.
+              !sym.isSynthetic &&
+                !sym.name.startsWith(nme.EVIDENCE_PARAM_PREFIX)
```

Comment:
if you're happy with the workaround, I would change the `FIXME` into `NOTE`. We tend to avoid `FIXME` comments as much as possible (with a few exceptions that point to specific GitHub issues)